blocks/edit/da-library/da-library.js

```diff
@@ -287,6 +294,22 @@ class DaLibrary extends LitElement {
     return { view, org, repo, ref: 'main', path: `/${path.join('/')}` };
   }
 
+  handlePreviewOpen(path, previewName) {
+    const previewPath = getPreviewUrl(path);
+    this._blockPreviewPath = previewPath || path;
+    this._previewItemName = previewName || '';
+  }
+
+  handlePreviewClose() {
+    this._blockPreviewPath = '';
+    this._previewItemName = '';
+  }
+
+  handlePreviewLoad() {
+    this.shadowRoot.querySelector('.da-fs-dialog-plugin').showModal();
+    this.shadowRoot.querySelector('.da-fs-dialog-plugin').classList.remove('hide');
```

Comment:
nit run selector once
```suggestion
    const fsDialogPlugin = this.shadowRoot.querySelector('.da-fs-dialog-plugin');
    fsDialogPlugin.showModal();
    fsDialogPlugin.classList.remove('hide');
```